common/achievement/index.ts

```diff
@@ -1,50 +1,57 @@
 import { prisma } from '../prisma';
 import { User } from '../user';
-import { EventValue } from './types';
-import { getMetricsByAction, getRelationByContext } from './util';
+import { isGamificationFeatureActive, getMetricsByAction } from './util';
 import { getLogger } from '../logger/logger';
 import { ActionID, SpecificNotificationContext } from '../notification/actions';
-import { NotificationContext } from '../notification/types';
-import { Achievement_event } from '../../graphql/generated';
-import { doesTemplateExistForAction } from './template';
+import { getTemplatesByAction } from './template';
+import { evaluateAchievement } from './evaluate';
+import { AchievementToCheck, ActionEvent, ConditionDataAggregations, UserAchievementContext, UserAchievementTemplate } from './types';
+import { createAchievement, getOrCreateUserAchievement } from './create';
+import { Achievement_template } from '../../graphql/generated';
+import { Prisma } from '@prisma/client';
+import { injectRecordValue, sortActionTemplatesToGroups } from './helper';
 
 const logger = getLogger('Achievement');
 
-export type ActionEvent = {
-    actionId: ActionID;
-    at: Date;
-    user: User;
-};
-export type Achievement_Event = {
-    userId?: string;
-    metric: string;
-    value: EventValue;
-    action?: string;
-    relation?: string; // e.g. "user/10", "subcourse/15", "match/20"
-};
 export async function actionTaken<ID extends ActionID>(user: User, actionId: ID, context: SpecificNotificationContext<ID>) {
```

Comment:
I'd go for rewardActionTaken, since the function is there to reward people whereas gamification would be the system itself. I'd also be fine with that. Tell me your thaughts.